### DISCUSSION TITLE
Can someone explain the difference in behavior between gcc and msvc in this scenario

### DISCUSSION THREAD

**Discussion by JasonDictos** [nlohmann/json#3061] [Q&A] [answered]
Here's the simple example source:

```
#include <cassert>
#include <nlohmann/json.hpp>
#include <variant>
using namespace nlohmann;
int main() {
    auto obj = json{{"bobo", "frodo"}};
    auto vobj = std::variant<json>{json{obj}};
    assert(obj.type() == std::get<0>(vobj).type());
```
Msvc 19.1 c++ latest, no assert https://godbolt.org/z/xaG7vY7sx 
Gcc 11.1 c++ 20, asserts (type array != type object)   https://godbolt.org/z/fdGb86cW6

**Comment by nlohmann** [comment, marked as answer]
Sure: https://json.nlohmann.me/home/faq/#brace-initialization-yields-arrays and #2583.
